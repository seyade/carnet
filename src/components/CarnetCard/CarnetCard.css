.carnet-card {
  font-weight: bold;
  background: #ebbc01;
  position: relative;
  width: 250px;
  /*border: 1px solid #bbb;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);*/
  border-radius: 2px;
  padding: 0 20px 20px;
  min-height: 200px;
  margin-right: 40px;
  margin-bottom: 40px;
  float: left;
  box-shadow: 0 0 5px 1px rgba(0, 0, 0, 0.1);
  z-index: 10;
}

.carnet-card:last-child {
  margin-right: 0;
}

.carnet-card a,
.carnet-card a:hover {
  color: #333;
  background: none;
  border: 0;
}

.carnet-card a:hover {
  text-decoration: underline;
}

.card-content-panel {
  border-bottom: 1px solid #555;
}

.carnet-card .card-remove-btn {
  position: absolute;
  top: -20px;
  left: 0;
  background: #ebbc01;
  padding: 5px 10px;
  box-shadow: 0 -2px 5px 0px rgba(0, 0, 0, 0.1);
  border-radius: 2px;
  z-index: 5;
}

.carnet-card .card-edit-btn {
  position: absolute;
  top: 0;
  right: -20px;
  background: #ebbc01;
  padding: 5px 10px;
  box-shadow: 3px 0 5px 0px rgba(0, 0, 0, 0.1);
  border-radius: 2px;
  z-index: 5;
}

.carnet-card .card-remove-btn:hover,
.carnet-card .card-edit-btn:hover {
  background: #ebbc01;
}

.card-title {
  margin-bottom: 20px;
}

.card-url {
  position: relative;
  padding-left: 5px;
  margin-bottom: 30px;
}

.card-url .fa-globe {
  position: absolute;
  left: -35px;
  top: -7px;
  font-size: 2.3em;
  padding: 3px 5px;
  background: #ebbc01;
  border-radius: 29px;
}

.card-info {
  border-top: 1px solid #777;
  padding: 20px 0 10px;
}
