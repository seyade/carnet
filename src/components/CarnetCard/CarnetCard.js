import React, { Component } from 'react';

class CarnetCard extends Component {

  static defaultProps = {
    bookmarkTitle: 'No data',
    url: '',
    info: 'Please add a Carnet bookmark.'
  }

  handleRemove() {
    alert('Card should be removed!');
  }

  handleShare(e) {
    e.preventDefault();
    alert('Card info can shared on social media.');
  }

  render() {
    return (
      <div className="card carnet-card col-md-4 col-xs-6">
        <a href="#" className="card-remove-btn" onClick={this.handleRemove}>Remove</a>
        <div>
          <h3 className="card-title">{this.props.carnet.bookmarkTitle}</h3>
          <p className="card-url"><a href={this.props.carnet.url} target="_blank">{this.props.carnet.url}</a></p>
          <p className="card-info">{this.props.carnet.info}</p>
        </div>
        <div>
          <a className="card-submit-btn btn btn-primary" onClick={this.handleShare}>Share</a>
        </div>
      </div>
    );
  }
}

export default CarnetCard;
