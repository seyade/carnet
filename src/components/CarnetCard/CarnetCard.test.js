import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import CarnetCard from '../CarnetCard/CarnetCard';

// it('renders CarnetCard without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<CarnetCard />, div);
// });

describe('CarnetCard', () => {
  let carnetcard,
    carnetData = {
      bookmarkTitle: 'Sabrina girls!!',
      url: 'http://www.sabrina.co',
      info: 'Some Sabrina best kitchen dishes'
    };

  beforeEach(() => {
    carnetcard = shallow(<CarnetCard carnet={carnetData} />);
  });


  it('renders as intended', () => {
    const component = renderer.create(<CarnetCard carnet={carnetData} />),
      json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  it('renders CarnetCard bookmark title', () => {
    expect(carnetcard.find('.card-title').text()).toEqual('Sabrina girls!!');
  });

  it('renders CarnetCard bookmark url or link', () => {
    expect(carnetcard.find('.card-url').text()).toEqual('http://www.sabrina.co');
  });

  it('renders CarnetCard bookmark info', () => {
    expect(carnetcard.find('.card-info').text()).toEqual('Some Sabrina best kitchen dishes');
  });

  it('should remove the card', () => {
    const handleRemove = jest.fn();

    expect(carnetcard.find('.card-remove-btn').length).toEqual(1);

    carnetcard.first().find('.card-remove-btn').simulate('click');

    expect(handleRemove).toHaveBeenCalled();
  });
});
