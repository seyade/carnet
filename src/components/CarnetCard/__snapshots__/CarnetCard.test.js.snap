exports[`CarnetCard renders as intended 1`] = `
<div
  className="card carnet-card col-md-4 col-xs-6">
  <a
    className="card-remove-btn"
    href="#"
    onClick={[Function]}>
    Remove
  </a>
  <div>
    <h3
      className="card-title">
      Sabrina girls!!
    </h3>
    <p
      className="card-url">
      <a
        href="http://www.sabrina.co"
        target="_blank">
        http://www.sabrina.co
      </a>
    </p>
    <p
      className="card-info">
      Some Sabrina best kitchen dishes
    </p>
  </div>
  <div>
    <a
      className="card-submit-btn btn btn-primary"
      onClick={[Function]}>
      Share
    </a>
  </div>
</div>
`;
